File: Two Pointer/p392_is_subsequence.py
class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:
        ps, pt=0,0

        if not s:
            return True

        while pt<len(t):
            if t[pt]==s[ps]:
                ps+=1
            
            if ps==len(s):
                return True
            
            pt+=1
        
        return False

#time complexity O(N)
#space complexity O(1)

File: Two Pointer/p392_is_subsequence.cpp
class Solution {
public:
  bool isSubsequence(string s, string t) {
    int ps = 0;

    for (int i = 0; i < t.size(); i++) {
      if (t[i] == s[ps]) {
        ps += 1;
      }
    }

    // cout << ps;
    // cout << s.size()-1;
    if (ps == s.size()) { // if it made it to end
      return true;
    }

    return false;
  }
};

// time complexity O(N)
// space complexity O(1)